scheduler-driver/src/test/java/gov/nasa/jpl/aerie/scheduler/RootfindingTest.java

```diff
@@ -164,7 +164,7 @@ public Duration valueAt(
         Duration.of(-2, Duration.SECONDS),
         twoSecond,
         100);
-    assertEquals(2, solution.history().getHistory().size());
-    assertEquals(new EquationSolvingAlgorithms.FunctionCoordinate<>(Duration.of(0, Duration.MICROSECONDS), Duration.of(0, Duration.MICROSECONDS)), solution.functionCoordinate());
+    assertEquals(1, solution.history().getHistory().size());
+    assertEquals(new EquationSolvingAlgorithms.FunctionCoordinate<>(Duration.of(-860925, Duration.MICROSECONDS), Duration.of(0, Duration.MICROSECONDS)), solution.functionCoordinate());
```

Comment:
Why did this value change? Additionally, why did the history _decrease_ in this test and _increase_ in the other test?

Reply:
it increased because the x1 value was out of the domain but still came back to the domain afterwards. But it still needed one more step to solve, it happens. You can go out of the x domain and then come back. 

For this one, the init value I am giving in this test is actually out of the x domain in the beginning. So instead of using the heuristic value, it uses another value randomly and finds the solution straight away.